### ISSUE TITLE
Build problem

### ISSUE THREAD

**Issue by rkimoakbioinformatics** [vv9k/docker-api-rs#9]
With "docker-api = { version = "*" }" in Cargo.toml of my project, `cargo build` gives

```
>cargo build                                                                                        
    Updating crates.io index
   Compiling docker-api v0.5.1
error[E0432]: unresolved import `futures_util::Stream`
  --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/docker-api-0.5.1/src/api/container/mod.rs:12:5
   |
12 |     Stream, TryStreamExt,
   |     ^^^^^^
   |     |
   |     no `Stream` in the root
   |     help: a similar name exists in the module (notice the capitalization): `stream`

error[E0432]: unresolved import `futures_util::Stream`
 --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/docker-api-0.5.1/src/api/system/mod.rs:9:20
  |
9 | use futures_util::{Stream, TryStreamExt};
  |                    ^^^^^^
  |                    |
  |                    no `Stream` in the root
  |                    help: a similar name exists in the module (notice the capitalization): `stream`

error[E0405]: cannot find trait `Stream` in crate `futures_util`
   --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/docker-api-0.5.1/src/builder.rs:539:33
    |
539 |           ) -> impl futures_util::Stream<Item = crate::Result<crate::conn::TtyChunk>> + Unpin + 'docker {
    |                                   ^^^^^^ not found in `futures_util`
    |
   ::: /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/docker-api-0.5.1/src/api/container/mod.rs:26:5
    |
26  | /     impl_api_ep! {container: Container, resp
27  | |         Inspect -> &format!("/containers/{}/json", container.id)
28  | |         Logs -> &format!("/containers/{}/logs", container.id)
29  | |         DeleteWithOpts -> &format!("/containers/{}", container.id), String, delete
30  | |     }
    | |_____- in this macro invocation
    |
    = note: this error originates in the macro `impl_api_ep` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
    |
5   | use futures_util::stream::Stream;
    |
5   | use std::stream::Stream;
    |
5   | use core::stream::Stream;
    |

Some errors have detailed explanations: E0405, E0432.
For more information about an error, try `rustc --explain E0405`.
error: could not compile `docker-api` due to 3 previous errors
```

Another thing is that `docker-api = { version = "0.6.0" }` did not work. Apparently, version 0.5.1 is the latest I could get with cargo build.

- Crate version: 0.6.0
- OS: Ubuntu 20.10
- Output of running `docker version` on the command line:
Client:
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.8
 Git commit:        20.10.7-0ubuntu5.1
 Built:             Mon Nov  1 00:33:40 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       20.10.7-0ubuntu5.1
  Built:            Thu Oct 21 23:58:58 2021
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          1.5.5-0ubuntu3
  GitCommit:        
 runc:
  Version:          1.0.1-0ubuntu2
  GitCommit:        
 docker-init:
  Version:          0.19.0
  GitCommit:

**Comment by vv9k** [comment]
Hmm, I can't really reproduce it in a clean project. I did the following:
- `cargo new --bin testdocker`
- Insert `docker-api = "*"` to `Cargo.toml`
- Add some code from docker api to main
- Correctly builds the version 0.6.0 without errors
- Set `docker-api = "0.5.1"` in `Cargo.toml`
- Build still works

So it seems like a local thing, because I'm also using this crate for other projects and it builds correctly. Could you perhaps post the full list of dependencies from `Cargo.toml` and do a `cargo clean` on the workspace?

One thing I'd also try is to delete the files cached by cargo in `/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/docker-api-0.5.1/` et al.

Also the error seems weird because there definitely is [`futures_util::Stream`](https://docs.rs/futures-util/0.3.17/futures_util/stream/trait.Stream.html) but in older versions like 0.2.0 [it wasn't available](https://docs.rs/futures-util/0.2.1/futures_util/index.html) so my guess is that somehow the older version gets imported and the new one doesn't erroring out.

**Comment by rkimoakbioinformatics** [comment]
Thanks for your reply. The dependencies in my Cargo.toml is as follows.
```
[dependencies]
lexical-core = { version = "0.7.5" }
tokio = {version = "*", features = ["full"]}
tokio-stream = { version = "*" }
warp = { version = "0.3" }
pretty_env_logger = { version = "*" }
lapin = { version = "*" }
lazy_static = { version = "*" }
log = { version = "0.4" }
http = { version = "*" }
hyper = { version = "*" }
dirs = { version = "*" }
substring = { version = "*" }
chrono = { version = "*" }
clap = { version = "*" }
async_once = { version = "*" }
common = { path = "../common" }
tokio-postgres = { version = "*" }
deadpool-postgres = { version = "0.7.0", features = ["config"] }
config = { version = "*" }
futures = { version = "*" }
serde = { version = ">=1.0.127", features = ["derive"] }
serde_json = { version = "*" }
serde_yaml = { version = "*" }
uuid = { version = "0.8.1", features = ["v4", "serde"] }
regex = { version = ">=1.5.4" }
#docker-api = { git = "https://github.com/vv9k/docker-api-rs", version = ">=0.6.0", features = ["swarm"] }
docker-api = { version = "*" }
```
I tried `cargo clean` followed by deleting `.cargo/registry/src/github.com-1ecc6299db9ec823/docker-api-0.5.1/` and `cargo build`, which still gave the same error.

At the end of the error message, there is
```
error[E0405]: cannot find trait `Stream` in crate `futures_util`
   --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/docker-api-0.5.1/src/builder.rs:539:33
    |
539 |           ) -> impl futures_util::Stream<Item = crate::Result<crate::conn::TtyChunk>> + Unpin + 'docker {
    |                                   ^^^^^^ not found in `futures_util`
    |
   ::: /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/docker-api-0.5.1/src/api/container/mod.rs:26:5
    |
26  | /     impl_api_ep! {container: Container, resp
27  | |         Inspect -> &format!("/containers/{}/json", container.id)
28  | |         Logs -> &format!("/containers/{}/logs", container.id)
29  | |         DeleteWithOpts -> &format!("/containers/{}", container.id), String, delete
30  | |     }
    | |_____- in this macro invocation
    |
    = note: this error originates in the macro `impl_api_ep` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
    |
5   | use futures_util::stream::Stream;
    |
5   | use std::stream::Stream;
    |
5   | use core::stream::Stream;
    |
```
Thus, it seems that re-exports from futures_util did not happen correctly. I tested removing docker-api from Cargo.toml, adding futures-util to Cargo.toml, and then adding `use futures_util::Stream;` to main.rs. `no 'Stream' in the root` error occurred. However, `use futures_util::stream::Stream;` passed `cargo check`.

I guess that it is not docker-api's problem then. But still, any help would be appreciated.

**Comment by vv9k** [comment]
Could you try out the following?
```toml
docker_api = { git = "https://github.com/vv9k/docker-api-rs", branch = "fix-futures-util" }
```
If it works I'll release a fixed version so you can use it directly rather than a git branch.

**Comment by rkimoakbioinformatics** [comment]
Thanks. With the branch, I got the following.
```
>cargo check
    Checking docker-api v0.6.0 (https://github.com/vv9k/docker-api-rs?branch=fix-futures-util#7073cefe)
error[E0599]: no method named `try_flatten_stream` found for opaque type `impl std::future::Future` in the current scope
   --> /home/user/.cargo/git/checkouts/docker-api-rs-f7f9028643fffcd3/7073cef/src/conn/transport.rs:193:14
    |
193 |             .try_flatten_stream()
    |              ^^^^^^^^^^^^^^^^^^ method not found in `impl std::future::Future`
    |
   ::: /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.8/src/future/try_future/mod.rs:532:8
    |
532 |     fn try_flatten_stream(self) -> TryFlattenStream<Self>
    |        ------------------ the method is available for `impl std::future::Future` here
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
3   | use futures_util::TryFutureExt;
    |

error[E0599]: no method named `try_flatten_stream` found for opaque type `impl std::future::Future` in the current scope
   --> /home/user/.cargo/git/checkouts/docker-api-rs-f7f9028643fffcd3/7073cef/src/conn/transport.rs:207:14
    |
207 |             .try_flatten_stream()
    |              ^^^^^^^^^^^^^^^^^^ method not found in `impl std::future::Future`
    |
   ::: /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.8/src/future/try_future/mod.rs:532:8
    |
532 |     fn try_flatten_stream(self) -> TryFlattenStream<Self>
    |        ------------------ the method is available for `impl std::future::Future` here
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
3   | use futures_util::TryFutureExt;
    |

error[E0599]: no method named `append_key_only` found for struct `url::form_urlencoded::Serializer` in the current scope
  --> /home/user/.cargo/git/checkouts/docker-api-rs-f7f9028643fffcd3/7073cef/src/util.rs:50:29
   |
50 |                         acc.append_key_only(k);
   |                             ^^^^^^^^^^^^^^^ method not found in `url::form_urlencoded::Serializer<'_, std::string::String>`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `docker-api` due to 3 previous errors
```
I found fixes for the two kinds of errors. Adding `use futures_util::TryFutureExt;` to transport.rs solved the first error. Adding `form_urlencoded = { version = "1.0.1" }` to my Cargo.toml solved the second error. Apparently, `append_key_only` exists in form_urlencoded::Serializer v1.0.1 but not in 1.0.0. 

After the two changes, my code compiled.

**Comment by vv9k** [comment]
I added `TryFutureExt` to `transport` and published a version `0.6.1`. Let me know if the issue is resolved :)

**Comment by rkimoakbioinformatics** [comment]
Thanks. It compiled!
